mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/inheritance/SingleTableInheritanceTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.test.inheritance;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Davide D'Alto
+ */
+public class SingleTableInheritanceTest extends OgmJpaTestCase {
+
+	Node parent;
+	Node simpleChild;
+	TextNode textChild;
+	NodeLink linkToSimple;
+	NodeLink linkToText;
+
+	@Before
+	public void setUp() {
+		EntityManager em = getFactory().createEntityManager();
+		try {
+			em.getTransaction().begin();
+
+			parent = new SimpleNode();
+			parent.setName( "root" );
+
+			simpleChild = new SimpleNode();
+			simpleChild.setName( "children 1" );
+
+			textChild = new TextNode();
+			textChild.setName( "children 2" );
+			textChild.setText( "a text" );
+
+			linkToSimple = new NodeLink();
+			linkToSimple.assignSource( parent );
+			linkToSimple.assignTarget( simpleChild );
+
+			linkToText = new NodeLink();
+			linkToText.assignSource( parent );
+			linkToText.assignTarget( textChild );
+
+			em.persist( parent );
+			em.persist( simpleChild );
+			em.persist( textChild );
+			em.persist( linkToText );
+			em.persist( linkToSimple );
+
+			em.getTransaction().commit();
+		}
+		finally {
+			em.close();
+		}
+	}
+
+	@Test
+	public void testSubclassReturnedByNativeQuery() throws Exception {
+		EntityManager em = getFactory().createEntityManager();
+		try {
+			em.getTransaction().begin();
+			@SuppressWarnings("unchecked")
+			List<TextNode> resultList = em.createNativeQuery( "db.Node.find({'DTYPE': 'TextNode'})", TextNode.class ).getResultList();
```

Comment:
Well, the expected return type is a parameter of the createNativeQuery method, so yes.

I'm not sure what happens if one doesn't specified the returned type. I think it returns an array of objects